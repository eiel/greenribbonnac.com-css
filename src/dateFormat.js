// @flow strict
const wdayMap = ["日", "月", "火", "水", "木", "金", "土"];

const dateFormat = (date: Date): string => {
  const month = date.getMonth() + 1;
  const day = date.getDate();
  const wday = wdayMap[date.getDay()];
  return `${month}月${day}日(${wday})`;
};

export default dateFormat;
