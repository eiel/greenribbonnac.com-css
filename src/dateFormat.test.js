// @flow strict
/* eslint-env jest */
import dateFormat from "./dateFormat";

it("dateFormat", () => {
  expect(dateFormat(new Date(2018, 9, 1))).toMatchSnapshot();
});
