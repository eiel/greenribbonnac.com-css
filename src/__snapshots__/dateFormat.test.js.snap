// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dateFormat 1`] = `"10月1日(月)"`;
